From 78a4258c52e545793be9eb7d7a06febcd5004162 Mon Sep 17 00:00:00 2001
From: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Date: Wed, 21 Sep 2022 15:43:48 +0200
Subject: [PATCH 3/7] Revert "vhost: fix missing virtqueue lock protection"

This reverts commit 78414da84eb5f5ae3594a073bdc978d3e14efeb4.

Please see issue https://github.com/spdk/spdk/issues/2518

Signed-off-by: Tomasz Zawadzki <tomasz.zawadzki@intel.com>
Change-Id: If475db9c14f3b52a280f18ee87251cbc6fdd9304
Reviewed-on: https://review.spdk.io/gerrit/c/spdk/dpdk/+/14601
Reviewed-by: Konrad Sztyber <konrad.sztyber@intel.com>
Reviewed-by: Ben Walker <benjamin.walker@intel.com>
---
 lib/vhost/vhost.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lib/vhost/vhost.c b/lib/vhost/vhost.c
index 24f94495c6..fde1cfdfb0 100644
--- a/lib/vhost/vhost.c
+++ b/lib/vhost/vhost.c
@@ -1299,15 +1299,11 @@ rte_vhost_vring_call(int vid, uint16_t vring_idx)
 	if (!vq)
 		return -1;
 
-	rte_spinlock_lock(&vq->access_lock);
-
 	if (vq_is_packed(dev))
 		vhost_vring_call_packed(dev, vq);
 	else
 		vhost_vring_call_split(dev, vq);
 
-	rte_spinlock_unlock(&vq->access_lock);
-
 	return 0;
 }
 
-- 
2.31.1

